stdlib/public/Concurrency/TaskCancellation.swift

```diff
@@ -96,10 +87,11 @@ extension Task {
     try await withTaskCancellationHandler(handler: handler, operation: operation)
   }
 
-  /// The default cancellation thrown when a task is cancelled.
+  /// The default error thrown by a cancelled task.
   ///
-  /// This error is also thrown automatically by `Task.checkCancellation()`,
-  /// if the current task has been cancelled.
+  /// The `Task.checkCancellation()` method throws this error
+  /// if the current task has been canceled.
+  /// You can use this error in your own cancellation-checking code also.
```

Comment:
If we wanted to we could document that it is allowed to throw other errors upon cancellation, but by convention returning this one is generally fine unless there are good reasons to use some other error.